src/compute-client/src/controller.rs

```diff
@@ -76,12 +75,12 @@ use crate::metrics::ComputeControllerMetrics;
 use crate::protocol::command::{ComputeParameters, PeekTarget};
 use crate::protocol::response::{PeekResponse, SubscribeBatch};
 
-mod instance;
 mod introspection;
 mod replica;
 mod sequential_hydration;
 
 pub mod error;
+pub mod instance;
```

Comment:
I would prefer if the `instance` module stays private and all calls go through the controller. This is a hole in the abstraction boundary otherwise.